src/app/components/pool/action/add/pool-create-basic-dialog.component.ts

```diff
@@ -38,6 +39,30 @@ export class PoolCreateBasicDialogComponent extends DynamicForm<Pool, PoolCreate
         return this._osControl.value && this._osControl.value.cloudServiceConfiguration;
     }
 
+    public get isSelectedImageDeprecated() {
+        if (this._osControl.value && this._osControl.value.virtualMachineConfiguration) {
+            const config = this._osControl.value.virtualMachineConfiguration;
+            const isDeprecatedImage = config && config.batchSupportEndOfLife;
+            if (isDeprecatedImage) {
```

Comment:
Instead of just checking for the presence of an end of life date, I think we should see how close it is and update the message accordingly. Let's just match the logic in the portal for this. I don't think it would be out of the question to have an end of life date defined that's 2 years in the future, but it wouldn't really be accurate to say it's "nearing" the end of life date in that case.

Reply:
Added new EOL messages to match those that are used in the Portal.